modules/gapi/src/streaming/onevpl/accelerators/accel_policy_dx11.cpp

```diff
@@ -63,53 +96,288 @@ void VPLDX11AccelerationPolicy::deinit(session_t session) {
-                                               surface_ptr_ctr_t creator) {
-    GAPI_LOG_DEBUG(nullptr, "pool size: " << pool_size << ", surface size bytes: " << surface_size_bytes);
+VPLDX11AccelerationPolicy::create_surface_pool(const mfxFrameAllocRequest& alloc_req,
+                                               mfxVideoParam& param) {
+    param.IOPattern = MFX_IOPATTERN_OUT_VIDEO_MEMORY;
 
-#ifdef CPU_ACCEL_ADAPTER
-    return adapter->create_surface_pool(pool_size, surface_size_bytes, creator);
-#endif
-    (void)pool_size;
-    (void)surface_size_bytes;
-    (void)creator;
-    throw std::runtime_error("VPLDX11AccelerationPolicy::create_surface_pool() is not implemented");
+    // allocate textures by explicit request
+    mfxFrameAllocResponse mfxResponse;
+    //TODO
+    mfxFrameAllocRequest alloc_request = alloc_req;
+    alloc_request.NumFrameSuggested = alloc_request.NumFrameSuggested;
+    mfxStatus sts = on_alloc(&alloc_request, &mfxResponse);
+    if (sts != MFX_ERR_NONE)
+    {
+        throw std::logic_error("Cannot create allocate memory for surfaces, error: " +
+                               mfxstatus_to_string(sts));
+    }
+
+    // get reference pointer
+    auto table_it = allocation_table.find(alloc_request.AllocId);
+    GAPI_DbgAssert (allocation_table.end() != table_it);
+
+    mfxU16 numSurfaces = alloc_request.NumFrameSuggested;
+
+    // NB: create pool with numSurfaces reservation
+    pool_t pool(numSurfaces);
+    for (int i = 0; i < numSurfaces; i++) {
+        std::unique_ptr<mfxFrameSurface1> handle(new mfxFrameSurface1 {});
+        handle->Info = param.mfx.FrameInfo;
+        handle->Data.MemId = mfxResponse.mids[i];
+
+        pool.push_back(Surface::create_surface(std::move(handle), table_it->second));
+    }
+
+    // remember pool by key
+    pool_key_t key = reinterpret_cast<pool_key_t>(table_it->second.get());
+    GAPI_LOG_INFO(nullptr, "New pool allocated, key: " << key <<
+                           ", surface count: " << pool.total_size());
+    try {
+        if (!pool_table.emplace(key, std::move(pool)).second) {
+            throw std::runtime_error(std::string("VPLDX11AccelerationPolicy::create_surface_pool - ") +
+                                     "cannot insert pool, table size: " + std::to_string(pool_table.size()));
+        }
+    } catch (const std::exception&) {
+        throw;
+    }
+    return key;
 }
 
 VPLDX11AccelerationPolicy::surface_weak_ptr_t VPLDX11AccelerationPolicy::get_free_surface(pool_key_t key)
 {
-#ifdef CPU_ACCEL_ADAPTER
-    return adapter->get_free_surface(key);
-#endif
-    (void)key;
-    throw std::runtime_error("VPLDX11AccelerationPolicy::get_free_surface() is not implemented");
+    auto pool_it = pool_table.find(key);
+    if (pool_it == pool_table.end()) {
+        std::stringstream ss;
+        ss << "key is not found: " << key << ", table size: " << pool_table.size();
+        const std::string& str = ss.str();
```

Comment:
You are using "Type& var" and "Type &var" from time to time across this file (and maybe others too)
It is a really minor suggestion, but you may consider to align it :)

Reply:
I hope it'll be fixed in the future